arch/arm/src/armv7-r/arm_cache.c

```diff
@@ -134,7 +134,15 @@ void up_invalidate_icache_all(void)
 
 void up_clean_dcache(uintptr_t start, uintptr_t end)
 {
-  cp15_clean_dcache(start, end);
+  if (cp15_cache_size() < (end - start))
+    {
+      cp15_clean_dcache(start, end);
+    }
+  else
+    {
+      cp15_clean_dcache_all();
+    }
+
   l2cc_clean(start, end);
 }
```

Comment:
let's optimize l2cc cache too

Reply:
l2cc have already implemention in l2cc_flush l2cc_clean function

Reply:
Nice!